runtime/src/main/java/org/corfudb/util/serializer/PrimitiveSerializer.java

```diff
@@ -40,12 +40,12 @@ public byte getType() {
     }
 
     @SuppressWarnings("unchecked")
-    private static Map<Class, SerializerFunction> getSerializerMap() {
+    private static Map<Class, Primitives> getSerializerMap() {
         ImmutableMap.Builder b = ImmutableMap.<Class, SerializerFunction>builder();
```

Comment:
Shouldn't this be of the type Primitives

Reply:
Thanks! Looks like the SuppressWarnings plus not assigning to a generic type was hiding that error.